packages/framework/azure-client/src/AzureClient.ts

```diff
@@ -73,29 +82,61 @@ export class AzureClient {
             config: {},
         });
 
-        const rootDataObject = await requestFluidObject<RootDataObject>(
+        const fluidContainer = await this.createFluidContainer(
             container,
-            "/",
+            this.props.connection,
         );
-        const createNewRequest = createAzureCreateNewRequest(
-            this.props.connection.orderer,
-            this.props.connection.storage,
-            this.props.connection.tenantId,
+        const services = this.getContainerServices(container);
+        return { container: fluidContainer, services };
+    }
+
+    /**
+     * Recreates new container out of specific version of another container
+     * @param id - Unique ID of the source container in Azure Fluid Relay.
+     * @param version - Unique version of the source container in Azure Fluid Relay.
+     * @param containerSchema - Container schema used to access data objects in the container.
+     * @returns New container instance along with associated services.
+     */
+    public async reCreateContainer(
```

Comment:
I'm thinking that "create from existing" is a better phrase than recreate. IMO recreate implies hydration of something that is dead. What this is really doing is creating a new container based off an existing one.

Reply:
Would developers ever use this outside of document recovery scenarios?  Going further on the naming, I think this should differentiate from `getContainer` more as well.  In a vacuum I'm not sure it's meaningfully different to the developer whether an existing container is "gotten" or "created", and something like `createContainerFromVersion` may be more informative

Reply:
Well, what's happening here is you are creating a new container from an existing container and version.

I want to separate out the scenario we are building the functionality for and the functionality itself. I could imagine developers creating a collection of "starter template containers that they then use to generate new containers. Maybe you could argue that this is not a good usage but it's another option.

Reply:
From our offline discussion let's go with `copyContainer` here to provide a more generically extensible api.